src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,305 @@
 
     this.options = ComponentOptions.initComponentOptions(element, QuerySuggestPreview, options);
 
+    if (!this.options.resultTemplate) {
+      this.logger.warn(
+        `Option "resultTemplate" is *REQUIRED* on the component "QuerySuggestPreview". The component or the search page might *NOT WORK PROPERLY*`,
+        `Check the following URL to create your result template`,
+        `https://developers.coveo.com/x/aIGfAQ`
+      );
+    }
+
     this.bind.onRootElement(OmniboxEvents.querySuggestGetFocus, (args: IQuerySuggestSelection) => this.querySuggestGetFocus(args));
   }
 
-  private get shouldShowPreviewResults() {
-    return this.options.numberOfPreviewResults > 0;
+  /**
+   * Builds and returns an array of HTMLElement with the given result set.
+   * @param results the result set to build an array of HTMLElement from.
+   */
+  public buildResults(results: IQueryResults): Promise<HTMLElement[]> {
+    const res: { elem: HTMLElement; idx: number }[] = [];
+    const resultsPromises = map(results.results, (result: IQueryResult, index: number) => {
+      return this.buildResult(result).then((resultElement: HTMLElement) => {
+        if (resultElement != null) {
+          res.push({ elem: resultElement, idx: index });
+        }
+        return resultElement;
+      });
+    });
+
+    // We need to sort by the original index order, because in lazy loading mode, it's possible that results does not gets rendered
+    // in the correct order returned by the index, depending on the time it takes to load all the results component for a given result template
+    return Promise.all(resultsPromises).then(() => {
+      return pluck(sortBy(res, 'idx'), 'elem');
+    });
+  }
+
+  /**
+   * Builds and returns an HTMLElement for the given result.
+   * @param result the result to build an HTMLElement from.
+   * @returns {HTMLElement}
+   */
+  public buildResult(result: IQueryResult): Promise<HTMLElement> {
+    Assert.exists(result);
+    QueryUtils.setStateObjectOnQueryResult(this.queryStateModel.get(), result);
+    QueryUtils.setSearchInterfaceObjectOnQueryResult(this.searchInterface, result);
+    return this.options.resultTemplate
+      .instantiateToElement(result, {
+        wrapInDiv: true,
+        checkCondition: true,
+        currentLayout: 'preview',
+        responsiveComponents: this.searchInterface.responsiveComponents
+      })
+      .then((resultElement: HTMLElement) => {
+        if (resultElement != null) {
+          Component.bindResultToElement(resultElement, result);
+        }
+        return this.autoCreateComponentsInsideResult(resultElement, result).initResult.then(() => {
+          return resultElement;
+        });
+      });
+  }
+
+  /**
+   * Reset the component when no suggestion are displayed
+   */
+  public handleNoSuggestion() {
+    clearTimeout(this.timer);
+    this.previousSuggestionHovered = null;
+  }
+
+  /**
+   * Create the container which the results preview will render
```

Comment:
Hmmm... not sure I understand that one:
Do you mean the container will contain all the results? Then I'd go with:
Creates the preview results container.